services/src/tpu_utils/tpu_service.rs

```diff
@@ -177,15 +178,23 @@ impl TpuService {
             }
         };
 
-        SolanaUtils::poll_slots(
-            self.rpc_client.clone(),
-            self.pubsub_client.clone(),
-            update_slot,
-        )
-        .await;
+        loop {
+            let Err(err) = SolanaUtils::poll_slots(
+                self.rpc_client.clone(),
+                &self.rpc_ws_address,
+                update_slot,
+            )
+            .await else {
+                bail!("current slot fetch task exited");
+            };
+
+            error!("slot fetch task error: {err:?}");
+
+            tokio::time::sleep(Duration::from_millis(2000)).await;
```

Comment:
We wait for slot to be polled for 2s and then again wait here for 2s so in total 4s whereas before we use to wait only for 2s. So remove this wait.